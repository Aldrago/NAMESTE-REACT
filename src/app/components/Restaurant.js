import { CDN_URL } from "../utils/constant";

const Restaurant = (props) => {
  const { name, cloudinaryImageId, sla, cuisines, avgRating } = props.resData;

  return (
    <div
      className="m-4 p-4 w-[300px] rounded-lg bg-gray-100 content-center"
      data-testid="resCard"
    >
      <img
        className="rounded-lg h-[200] w-[280]"
        src={CDN_URL + cloudinaryImageId}
      />
      <h2 className="pt-2 font-bold truncate">{name}</h2>
      <h6 className="py-1 text-xs truncate">{cuisines.join(", ")}</h6>
      <h4 className="pt-2">{sla.deliveryTime} minutes</h4>
      <h4>‚≠ê{avgRating}</h4>
    </div>
  );
};

export const withPromotedLabel = (Restaurant) => {
  return (props) => {
    return (
      <div>
        <label className="absolute bg-black text-white p-1 rounded-lg bg-opacity-[0.7] ">
          Promoted
        </label>
        <Restaurant {...props} />
      </div>
    );
  };
};

export default Restaurant;
