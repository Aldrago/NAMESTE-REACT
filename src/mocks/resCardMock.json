{
  "id": "60036",
  "name": "Havmor Havfunn Ice Cream",
  "cloudinaryImageId": "49845b494b1e637209b7bd1c7ca184ad",
  "locality": "Sector 43",
  "areaName": "Sushant Lok",
  "costForTwo": "₹400 for two",
  "cuisines": ["Ice Cream", "Ice Cream Cakes", "Desserts", "Beverages"],
  "avgRating": 4.7,
  "parentId": "94570",
  "avgRatingString": "4.7",
  "totalRatingsString": "1K+",
  "sla": {
    "deliveryTime": 31,
    "lastMileTravel": 7.6,
    "serviceability": "SERVICEABLE",
    "slaString": "30-35 mins",
    "lastMileTravelString": "7.6 km",
    "iconType": "ICON_TYPE_EMPTY"
  },
  "availability": {
    "nextCloseTime": "2024-05-26 00:00:00",
    "opened": true
  },
  "badges": {},
  "isOpen": true,
  "type": "F",
  "badgesV2": {
    "entityBadges": {
      "imageBased": {},
      "textBased": {},
      "textExtendedBadges": {}
    }
  },
  "aggregatedDiscountInfoV3": {
    "header": "₹100 OFF",
    "subHeader": "ABOVE ₹399",
    "discountTag": "FLAT DEAL"
  },
  "differentiatedUi": {
    "displayType": "ADS_UI_DISPLAY_TYPE_ENUM_DEFAULT",
    "differentiatedUiMediaDetails": {
      "mediaType": "ADS_MEDIA_ENUM_IMAGE",
      "lottie": {},
      "video": {}
    }
  },
  "reviewsSummary": {},
  "displayType": "RESTAURANT_DISPLAY_TYPE_DEFAULT",
  "restaurantOfferPresentationInfo": {}
}
