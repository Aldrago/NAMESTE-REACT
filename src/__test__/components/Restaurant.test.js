import { render } from "@testing-library/react";
import Restaurant, { withPromotedLabel } from "../../app/components/Restaurant";
import resCardMock from "../../mocks/resCardMock.json";
import "@testing-library/jest-dom"

describe("Restaurant card component test cases", () => {
  test("Should match Restaurant snapshot", () => {
    const { container } = render(<Restaurant resData={resCardMock} />);
    expect(container).toMatchSnapshot();
  });

  test("Should match Restaurant with promoted label snapshot", () => {
    const RestaurantCardsPromoted = withPromotedLabel(Restaurant);
    const { container } = render(
      <RestaurantCardsPromoted resData={resCardMock} />
    );
    expect(container).toMatchSnapshot();
  });
});
