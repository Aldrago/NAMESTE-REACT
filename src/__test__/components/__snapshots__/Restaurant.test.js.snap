// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurant card component test cases Should match Restaurant snapshot 1`] = `
<div>
  <div
    class="m-4 p-4 w-[300px] rounded-lg bg-gray-100 content-center"
    data-testid="resCard"
  >
    <img
      class="rounded-lg h-[200] w-[280]"
      src="https://media-assets.swiggy.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_660/49845b494b1e637209b7bd1c7ca184ad"
    />
    <h2
      class="pt-2 font-bold truncate"
    >
      Havmor Havfunn Ice Cream
    </h2>
    <h6
      class="py-1 text-xs truncate"
    >
      Ice Cream, Ice Cream Cakes, Desserts, Beverages
    </h6>
    <h4
      class="pt-2"
    >
      31
       minutes
    </h4>
    <h4>
      ⭐
      4.7
    </h4>
  </div>
</div>
`;

exports[`Restaurant card component test cases Should match Restaurant with promoted label snapshot 1`] = `
<div>
  <div>
    <label
      class="absolute bg-black text-white p-1 rounded-lg bg-opacity-[0.7] "
    >
      Promoted
    </label>
    <div
      class="m-4 p-4 w-[300px] rounded-lg bg-gray-100 content-center"
      data-testid="resCard"
    >
      <img
        class="rounded-lg h-[200] w-[280]"
        src="https://media-assets.swiggy.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_660/49845b494b1e637209b7bd1c7ca184ad"
      />
      <h2
        class="pt-2 font-bold truncate"
      >
        Havmor Havfunn Ice Cream
      </h2>
      <h6
        class="py-1 text-xs truncate"
      >
        Ice Cream, Ice Cream Cakes, Desserts, Beverages
      </h6>
      <h4
        class="pt-2"
      >
        31
         minutes
      </h4>
      <h4>
        ⭐
        4.7
      </h4>
    </div>
  </div>
</div>
`;
